src/components/QueryEditor.tsx

```diff
@@ -70,7 +72,19 @@ export function QueryEditor({ query, onChange, onRunQuery, datasource }: Props)
     });
   };
 
-  
+  const addPlaceholder = (editor: any, monaco: any) => {
+    editor.executeEdits(null, [{ range: new monaco.Range(1, 1, 1, 1), text: placeholder }]);
+    editor.onDidFocusEditorText(() => {
+      if (editor.getValue() === placeholder) {
+        editor.executeEdits(null, [{ range: new monaco.Range(1, 1, placeholder.length, 1), text: '' }]);
+      }
+    });
+    editor.onDidBlurEditorText(() => {
+      if (editor.getValue() === '') {
+        editor.executeEdits(null, [{ range: new monaco.Range(1, 1, 1, 1), text: placeholder }]);
+      }
+    });
+  };
```

Comment:
Do you think that the text disappearing on focus is unexpected? 🤔
IMO since it is general guidance and not something users would refer to while writing it's good as is, but wanted to validate with you too.

the behavior from HTML input placeholders came to mind, what do you think about hiding the placeholder when the user types something? (IMO, way harder to implement in a way that feels good for the user and probably not worth it, but wanted to throw the idea here)

Reply:
HTML-placeholder-like would def be nicer, but harder to implement because that probably needs to be a pseudo-element with click-through so the cursor is at the right position.